coderd/coderd.go

```diff
@@ -950,6 +950,8 @@ func New(options *Options) *API {
 				r.Get("/watch", api.watchWorkspace)
 				r.Put("/extend", api.putExtendWorkspace)
 				r.Put("/dormant", api.putWorkspaceDormant)
+				r.Put("/favorite", api.putFavoriteWorkspace)
+				r.Delete("/favorite", api.deleteFavoriteWorkspace)
```

Comment:
nit: as we modify a single record in `workspaces`, I would consider single PATCH operation. This form is also correct but longer.

Reply:
What you say is correct. One can also make an argument for the simplicity of a handler that does one thing only instead of a complex handler that maybe does many things. I don't feel particularly strongly either way; the separate handler approach seems to be more in line with existing conventions.